### PR TITLE
1.1.0.0

### PR DESCRIPTION
**Pull request by seitenca**

### CODE REVIEW

**Comment by mitchej123** [comment]
Dream - let me and moron review this before merging

File: battlegear mod src/minecraft/assets/battlegear2/lang/en_US.lang [hunk 1]
```diff
@@ -112,6 +113,7 @@ attribute.quiver.arrow.empty=Empty
 attribute.quiver.arrow.count=Arrows
 
 lore.base.arrow.explosive=Packs some explosive power
+lore.base.arrow.ice=Effective agains ? ice blaze?
```
Comment (richardhendricks) [inline battlegear mod src/minecraft/assets/battlegear2/lang/en_US.lang]: Change this to 
Effective against hot creatures like blazes and lava monsters

File: battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        isImmuneToFire = false;
+    }
+
+    public EntityIceArrow(World par1World, EntityLivingBase par2EntityLivingBase, EntityLivingBase par3EntityLivingBase, float par4, float par5) {
+        super(par1World, par2EntityLivingBase, par3EntityLivingBase, par4, par5);
+        isImmuneToFire = false;
+    }
+
+    @Override
+    public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
```
Comment (richardhendricks) [inline battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: You can insert a custom damage source here as discussed in the ticket

Similar to the piercing arrow, except use "Frost" as the damagesource name. 
https://github.com/GTNewHorizons/Battlegear2/blob/7e8dcdb2811f1c55f892925756f4924e2e65a529/battlegear%20mod%20src/minecraft/mods/battlegear2/items/arrows/EntityPiercingArrow.java#L51

**Review by richardhendricks — CHANGES_REQUESTED** [review]

File: battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java (hunk 2 above)
Comment (seitenca) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: @Override
	public boolean onHitEntity(Entity entityHit, DamageSource Frost, float ammount) {
		if(entityHit.isImmuneToFire() == true) {
			entityHit.attackEntityFrom(getFrostDamage(), ammount);
			if(!worldObj.isRemote && entityHit instanceof IShearable){
				if(((IShearable)entityHit).isShearable(null, worldObj, (int) entityHit.posX, (int) entityHit.posY, (int) entityHit.posZ)){
					ArrayList<ItemStack> drops = ((IShearable)entityHit).onSheared(null, worldObj, (int) entityHit.posX, (int) entityHit.posY, (int) entityHit.posZ, 1);
					Random rand = new Random();
	                for(ItemStack stack : drops){
	                    EntityItem ent = entityHit.entityDropItem(stack, 1.0F);
	                    ent.motionY += rand.nextFloat() * 0.05F;
	                    ent.motionX += (rand.nextFloat() - rand.nextFloat()) * 0.1F;
	                    ent.motionZ += (rand.nextFloat() - rand.nextFloat()) * 0.1F;
	                }
				}
			}
		
		}
		return true;
	}
something like this?

Comment (richardhendricks) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: More like this

```java 
	public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
		entityHit.attackEntityFrom(new EntityDamageSourceIndirect("frost", null, shootingEntity).setProjectile().setDamageBypassesArmor(), ammount);
		return true;
	}
```

Comment (richardhendricks) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: It probably should apply some slow effect, and maybe ice particles on impact but we can add those later.

Comment (seitenca) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: entity that != isImmuneToFire, are inmune to these arrows.
We leave it that way or I change it

Comment (seitenca) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: @Override
	public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
		boolean flag = false;
		if(entityHit.isImmuneToFire() == true) {
			entityHit.attackEntityFrom(new EntityDamageSourceIndirect("frost", null, shootingEntity).setProjectile().setDamageBypassesArmor(), ammount);
		}
		if(entityHit instanceof EntityLivingBase){
		((EntityLivingBase) entityHit).addPotionEffect(new PotionEffect(Potion.moveSlowdown.id, 200));
	        ((EntityLivingBase) entityHit).setArrowCountInEntity(((EntityLivingBase) entityHit).getArrowCountInEntity()+1);
		flag = true;
		}
		setDead();
		return flag;
	}

Comment (richardhendricks) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: Looks good to me.  What's going on with the setDead();? Is that to "kill" the arrow?

Comment (seitenca) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: @Override
	public void onHitGround(int x, int y, int z) {
		setDead();
	}

Comment (seitenca) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: @Override
	public boolean onHitEntity(Entity entityHit, DamageSource source, float ammount) {
		boolean flag = false;
		if(entityHit.isImmuneToFire() == true) {
		entityHit.attackEntityFrom(new EntityDamageSourceIndirect("frost", null, shootingEntity).setProjectile().setDamageBypassesArmor(), 13);
		}
		if(entityHit instanceof EntityLivingBase){
		((EntityLivingBase)entityHit).attackEntityFrom(new EntityDamageSourceIndirect("default", null, shootingEntity).setProjectile(), 3);
		((EntityLivingBase)entityHit).addPotionEffect(new PotionEffect(Potion.moveSlowdown.id, 200, 2));
	        ((EntityLivingBase)entityHit).setArrowCountInEntity(((EntityLivingBase) entityHit).getArrowCountInEntity()+1);
		flag = true;
		}
		setDead();
		return flag;
	}

Comment (richardhendricks) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: Wait, does this mean someone who drinks a resistance to fire potion takes extra damage from the arrow?  
...
I'll allow it. :D

Comment (seitenca) [reply battlegear mod src/minecraft/mods/battlegear2/items/arrows/EntityIceArrow.java]: yes hehe
13 damage if is immune else 3
what do you think?

**Comment by richardhendricks** [comment]
I still need to do some testing, but I think the extra damage is working against special mob blazes.

**Comment by Dream-Master** [comment]
@richardhendricks @mitchej123 ready to merge ?

**Review by Dream-Master — APPROVED** [review]

**Comment by Dream-Master** [comment]
DreamMasterXXLToday at 10:51 AM
morning
can this merged ?
https://github.com/GTNewHorizons/Battlegear2/pull/1

moronwmachinegunToday at 3:15 PM
Yes

**Pull request merged**